Frontend-v1-Original/components/ssRewards/lib/queries.ts

```diff
@@ -151,81 +148,161 @@ export const getRewardBalances = async (
     }
   }
 
-  const rewardsCalls = filteredPairs2.map((pair) => {
-    return {
-      address: pair.gauge.address,
-      abi: CONTRACTS.GAUGE_ABI,
-      functionName: "earned",
-      args: [CONTRACTS.GOV_TOKEN_ADDRESS, address],
-    } as const;
-  });
+  const tokens = getInitBaseAssets();
 
-  const oBlotrRewardsCalls = filteredPairs2.map((pair) => {
-    return {
-      address: pair.gauge.address,
-      abi: CONTRACTS.GAUGE_ABI,
-      functionName: "earned",
-      args: [CONTRACTS.OPTION_TOKEN_ADDRESS, address],
-    } as const;
+  await Promise.all(
+    filteredPairs2.map(async (pair) => {
+      const gaugeAddress = pair.gauge.address;
+      const rewardNumber = await viemClient.readContract({
+        address: gaugeAddress,
+        abi: CONTRACTS.GAUGE_ABI,
+        functionName: "rewardsListLength",
+      });
+      const rewardTokens: {
+        address: `0x${string}`;
+        symbol: string;
+        decimals: number;
+      }[] = [];
+      const arr = Array.from({ length: Number(rewardNumber) }, (_, i) => i);
```

Comment:
```suggestion
      const arr = [...Array(Number(rewardNumber)).keys()]
```
Want to try something more modern?

Reply:
kewl